src/core/libmaven/mzUtils.cpp

```diff
@@ -593,93 +424,92 @@ Series:  Prentice-Hall Series in Automatic Computation
+            x2 += x[i] * x[i];
+            y2 += y[i] * y[i];
         }
         if (n == 0) return 0;
-        double var1 = x2-(sumx*sumx)/n;
-        double var2 = y2-(sumy*sumy)/n;
+        double var1 = x2 - (sumx * sumx) / n;
+        double var2 = y2 - (sumy * sumy) / n;
         if ( var1 == 0 || var2 == 0 ) return 0;
-        return (sumxy -( sumx*sumy)/n) / sqrt((x2-(sumx*sumx)/n)*(y2-(sumy*sumy)/n));
+        float corr = (sumxy -( sumx * sumy) / n) /
+                       sqrt((x2 - (sumx * sumx) / n) *
+                          (y2 - (sumy * sumy) / n));
+        return corr;
     }
 
-
-    /*peak fitting function*/
-    void gaussFit(const vector<float>&ycoord, float* sigma, float* R2) {
-
+    pair<float, float> gaussFit(const vector<float>& ycoord)
+    {
         float s = 20;
         float min_s = 0;
         float minR = 1e99;
 
         //find best fit
-        if (ycoord.size()<3) return;
-        vector<float>yobs=ycoord;
-        int ysize=yobs.size();
-        int midpoint  = int(ysize/2);
+        if (ycoord.size()<3){
+            pair<float, float> res = make_pair(numeric_limits<float>::max(),
+                                               numeric_limits<float>::max());
+            return res;
+        }
+        vector<float>yobs = ycoord;
+        int ysize = yobs.size();
+        int midpoint = int(ysize/2);
         //find maximum point ( assuming it somewhere around midpoint of the yobs);
-        float ymax = max(max(yobs[midpoint], yobs[midpoint-1]),yobs[midpoint+1]);
+        float ymax = max(max(yobs[midpoint], yobs[midpoint-1]),
+                         yobs[midpoint+1]);
         float ymin = min( yobs[0], yobs[ysize-1]);
 
-        //initialize x vector, values centered around 0, forxample  -2, -1, 0, 1, 2
-        int xinit = int(ysize/2)*-1;
-        vector<float>x(ysize,0);
-        int greaterZeroCount=0;
-
-        for(int i=0; i<ysize; i++ ) {
-            x[i] = xinit+i;
-            if ( yobs[i] > ymin ) greaterZeroCount++;
-            yobs[i] = (yobs[i]-ymin)/(ymax-ymin);
-            if(yobs[i]<0) yobs[i]=0;
+        //initialize x vector, values centered around 0,
+        //forxample  -2, -1, 0, 1, 2
+        int xinit = int(ysize / 2) * -1;
+        vector<float> x(ysize, 0);
+        int greaterZeroCount = 0;
+
+        for(int i = 0; i < ysize; i++){
+            x[i] = xinit + i;
+            if (yobs[i] > ymin) greaterZeroCount++;
+            yobs[i] = (yobs[i] - ymin) / (ymax - ymin);
+            if(yobs[i] < 0) yobs[i] = 0;
         }
 
-        /*
-           cerr << "fitting yobs:" << endl; for(int i=0; i < ysize; i++ )
-           cerr << setprecision(2) << yobs[i] << ", "; cerr << endl;
-           */
-
         bool converged = false;
         int ittr = 0;
 
-        if (greaterZeroCount <= 3 ) return;
+        if (greaterZeroCount <= 3 ){
+            pair<float, float> res = make_pair(FLT_MAX,
+                                               FLT_MAX);
```

Comment:
I see you have changed the above lines to use `numeric_limits` but not here. In general, most suggestions from reviewers should apply to the whole change-set, unless it is something specific. So, what I usually do when I get a review suggestion, is that I would look for other instances (that were not addressed by the reviewer) where that suggestion can be applied and fix those too.

We can let this one be as it is, but you can make this a practice in future PRs.